### PR TITLE
Obtain an immutable array from a mutable one

### PR DESCRIPTION
**Pull request by rjolly**

### CODE REVIEW

File: library/src/scala/IArray.scala [hunk 1]
```diff
@@ -268,6 +268,9 @@ object IArray {
   // A convenience to avoid having to cast everything by hand
   private given [A] as Conversion[Array[A], IArray[A]] = identity[Sub[A]]
 
+  /** Obtain an immutable array from a mutable one. */
+  def apply[T](s: Array[T]): IArray[T] = s
```
Comment (smarter) [inline library/src/scala/IArray.scala:272]: This is unsafe, so it should be called something like `unsafeFromArray`

File: library/src/scala/IArray.scala [hunk 2]
```diff
@@ -268,6 +268,9 @@ object IArray {
   // A convenience to avoid having to cast everything by hand
   private given [A] as Conversion[Array[A], IArray[A]] = identity[Sub[A]]
 
+  /** Obtain an immutable array from a mutable one. */
```
Comment (smarter) [inline library/src/scala/IArray.scala:271]:
```suggestion
  /** Convert an array into an immutable array without copying, the original array
   *   must _not_ be mutated after this or the guaranteed immutablity of IArray will
   *   be violated.
   */
```

**Comment by sjrd** [comment]
I don't think we should have this at all. We don't have `List.unsafeFromListBuffer`, so why should we have `IArray.unsafeFromArray`?

**Comment by smarter** [comment]
But we do have `ArraySeq.unsafeWrapArray`. I think it makes sense in the context of Java interop where you're getting arrays from Java which are documented as immutable, in the same way that all values coming from Java are nullable but we still provide `.nn` under `-Yexplicit-nulls`.

**Comment by rjolly** [comment]
`List.unsafeFromListBuffer` would be useful to me. In Java we have `Collections.unmodifiableCollection(c)` and friends.

**Comment by sjrd** [comment]
Java never encodes immutability in the type system, so `unmodifiableX` is not an unsafe operation. It's using `j.u.List` that's unsafe to start with.

**Pull request merged**